laniakea/core/providers/ec2/manager.py

```diff
@@ -59,22 +60,32 @@ def connect(self, region, **kw_params):
         self.ec2 = boto.ec2.connect_to_region(region, **kw_params)
         if not self.ec2:
             raise Exception('Unable to connect to region "%s"' % region)
+        self.remote_images.clear()
 
-        if self.images:
-            # Resolve AMI names in our configuration to their IDs
-            logger.info('Retrieving available AMIs...')
-            remote_images = self.ec2.get_all_images(owners=['self', 'amazon', 'aws-marketplace'])
-            for i in self.images:
-                if "image_name" in self.images[i] and 'image_id' not in self.images[i]:
-                    image_name = self.images[i]['image_name']
-                    for ri in remote_images:
-                        if ri.name == image_name:
-                            if 'image_id' in self.images[i]:
-                                raise Exception('Ambiguous AMI name "%s" resolves to multiple IDs' % image_name)
-                            self.images[i]['image_id'] = ri.id
-                            del self.images[i]['image_name']
-                    if 'image_id' not in self.images[i]:
-                        raise Exception('Failed to resolve AMI name "%s" to an AMI ID' % image_name)
+        if self.images and any(('image_name' in img and 'image_id' not in img) for img in self.images.values()):
```

Comment:
Can we first make `'image_name' in img and 'image_id' not in img` into a boolean?

I wasn't sure what I was looking at until I realised they were the same thing.